cli/cache/mod.rs

```diff
@@ -74,6 +80,17 @@ impl Loader for FetchCacher {
     specifier: &ModuleSpecifier,
     is_dynamic: bool,
   ) -> LoadFuture {
+    if specifier.scheme() == "npm" {
+      return Box::pin(futures::future::ready(
+        match npm::NpmPackageReference::from_specifier(specifier) {
+          Ok(_) => Ok(Some(deno_graph::source::LoadResponse::External {
+            specifier: specifier.clone(),
+          })),
+          Err(err) => Err(err),
+        },
+      ));
+    }
```

Comment:
1. Npm package references are maked as "external" so deno graph doesn't analyze them.
2. These npm package references are analyzed and an "npm install" essentially occurs behind the scenes.
3. The CliModuleLoader via `ps.npm_resolver` then does most of the work for resolving specifiers in npm packages.